### ISSUE TITLE
CI and Upstream CI failures

### ISSUE THREAD

**Issue by anissa111** [UXARRAY/uxarray#207]
Summary of issues:
- Upstream CI
    - pip is causing some dependency clashes since we use pip to install upstream packages after solving a conda environment
- Nightly CI
    - It seems like the UMass tredds server is... down? And has been for at least two weeks. The `test_read_ugrid_opendap` test fails because it cannot connect, not because anything on our side is broken. 
    - We're getting a [warning](https://github.com/UXARRAY/uxarray/actions/runs/4001972562/jobs/6868752083#step:6:92). Its not breaking anything currently, but we should probably address it
```
Encountered the use of a type that is scheduled for deprecation: type 'reflected list' found for argument 'node_coord' of function '_convert_node_lonlat_rad_to_xyz'.
```

**Comment by rajeeja** [comment]
How do you plan to address it? Disable it or let it fail gracefully or something else?

**Comment by anissa111** [comment]
> How do you plan to address it? Disable it or let it fail gracefully or something else?

I was going to ask about that in my PR. I currently have a branch that is just try/catching the url before trying tests and printing a warning if it can't find the file, but ideally I think it would be good to do something like
1. find an alternative OPeNDAP url
2. include a backup OPeNDAP url in case the first one goes down for maintenance or something
3. print a warning that the test didn't run if, by some chance, both OPeNDAP sources fail

**Comment by rajeeja** [comment]
3 sounds good.

**Comment by anissa111** [comment]
Yes, the only problem is that I don't know any other two ugrid file on separate OPeNDAP sources. Any ideas there?

**Comment by rljacob** [comment]
What are we trying to get from UMass?

**Comment by dcherian** [comment]
Can you vendor or generate some "nice" small files for testing? It seems like a bad idea to depend on remote endpoints that you don't control.

**Comment by anissa111** [comment]
@rljacob, here is the test. It was contributed by @ocefpaf in https://github.com/UXARRAY/uxarray/pull/68. 

``` python
def test_read_ugrid_opendap(self):
    """Read an ugrid model from an OPeNDAP URL."""

    url = "http://www.smast.umassd.edu:8080/thredds/dodsC/FVCOM/NECOFS/Forecasts/NECOFS_GOM3_FORECAST.nc"
    xr_grid = xr.open_dataset(url, drop_variables="siglay")
    ugrid = ux.Grid(xr_grid)
    assert isinstance(getattr(ugrid, "Mesh2_node_x"), xr.DataArray)
    assert isinstance(getattr(ugrid, "Mesh2_node_y"), xr.DataArray)
    assert isinstance(getattr(ugrid, "Mesh2_face_nodes"), xr.DataArray)
```

**Comment by ocefpaf** [comment]
> @rljacob, here is the test. It was contributed by @ocefpaf in #68.

We can find a more stable dataset to test and use pytest-vcr to "record" it. Probably better than making this. I can to that but for now I would suggest to just mark all test that required a connection as "expected fail" or something similar to make reviews a bit easier.
